File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
class Solution {
    public int solution(int[] arr) {
        int answer = 0;
        
        while(true){
            int check = 0;
            for(int i = 0; i < arr.length; i++){
                if(arr[i] >= 50 && arr[i] % 2 == 0){
                    arr[i] /= 2;
                }else if(arr[i] < 50 && arr[i] % 2 != 0){
                    arr[i] = arr[i] * 2 + 1;
                }else {
                    check++;
                }
            }
            answer++;
            
            if(check == arr.length) break;
        }
        return answer - 1;
    }
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// arr_len은 배열 arr의 길이입니다.
int solution(int arr[], size_t arr_len) {
    int answer = 0;
    int count = 0;
    
    while(1){
        count = 0;
        for(int i = 0; i < arr_len; i++){
            if(arr[i] >= 50 && arr[i] % 2 == 0){
                arr[i] /= 2;
            }else if(arr[i] < 50 && arr[i] % 2 != 0){
                arr[i] *= 2;
                arr[i] += 1;
            }else {
                count++;
            }
        }
        
        if(count == arr_len){
            return answer;
        }else {
            answer++;
        }
    }
    
    return -1;
}